diff --git a/Makefile b/Makefile
index 4d0784b..29f21ad 100644
--- a/Makefile
+++ b/Makefile
@@ -106,10 +106,11 @@ dest_mandir := $(DESTDIR)$(MANDIR)
 dest_docdir := $(DESTDIR)$(DOCDIR)
 dest_bindir := $(DESTDIR)$(BINDIR)
 dest_libdir := $(DESTDIR)$(LIBDIR)
+dest_sitedir := $(DESTDIR)$(SITEDIR)
 
 install: all
-	$(INSTALL) -d $(dest_bindir) \
-		$(dest_libdir)/bup $(dest_libdir)/cmd \
+	$(INSTALL) -d $(dest_bindir) $(dest_libdir) \
+		$(dest_sitedir)/bup $(dest_libdir)/cmd \
 		$(dest_libdir)/web $(dest_libdir)/web/static
 	test -z "$(man_roff)" || install -d $(dest_mandir)/man1
 	test -z "$(man_roff)" || $(INSTALL) -m 0644 $(man_roff) $(dest_mandir)/man1
@@ -122,13 +123,14 @@ install: all
 	done
 	cd "$(dest_bindir)" && \
 	  ln -sf "$$($(bup_python) -c 'import os; print(os.path.relpath("$(abspath $(dest_libdir))/cmd/bup"))')"
+	sed -i "/^_libdir = / s, = .\+, = b'$(LIBDIR)'," lib/bup/path.py
 	set -e; \
 	$(INSTALL) -pm 0644 \
 		lib/bup/*.py \
-		$(dest_libdir)/bup
+		$(dest_sitedir)/bup
 	$(INSTALL) -pm 0755 \
 		lib/bup/*$(SOEXT) \
-		$(dest_libdir)/bup
+		$(dest_sitedir)/bup
 	$(INSTALL) -pm 0644 \
 		lib/web/static/* \
 		$(dest_libdir)/web/static/
@@ -137,10 +139,10 @@ install: all
 		$(dest_libdir)/web/
 	if test -e lib/bup/checkout_info.py; then \
 	    $(INSTALL) -pm 0644 lib/bup/checkout_info.py \
-	        $(dest_libdir)/bup/source_info.py; \
+	        $(dest_sitedir)/bup/source_info.py; \
 	else \
 	    ! grep -qF '$$Format' lib/bup/source_info.py; \
-	    $(INSTALL) -pm 0644 lib/bup/source_info.py $(dest_libdir)/bup/; \
+	    $(INSTALL) -pm 0644 lib/bup/source_info.py $(dest_sitedir)/bup/; \
 	fi
 
 config/config.h: config/config.vars
